app/views/recipes/show.html.erb

```diff
@@ -0,0 +1,53 @@
+<div class="container">
+  <h1><%= @recipe.name %></h1>
+  <div class="genral-detal">
+    <div class="details">
+      <h3>Preparation time: <%= @recipe.preparation_time %></h3>
+      <h3>Cooking time: <%= @recipe.cooking_time %></h3>
+      <h3><%= @recipe.description %></h3>
+    </div>
+    <div>
+      <%= form_with(model: @recipe, url: user_recipe_path(@user, @recipe), local: true, method: :patch) do |f| %>
+        <div>
+          <p><%= f.label :public %>
+            <label class="switch">
+              <%= f.hidden_field :public, value: false %>
+              <%= f.check_box :public, { onchange: 'this.form.submit()',
+                                        disabled: @user != current_user }, true, false %>
+              <span class="slider"></span>
+            </label>
+          </p>
+        </div>
+      <% end %>
+    </div>
+  </div>
+
+  <div class="btns">
+    <%= button_to "Generate Shopping List", "", class: "add-food-link"%>
```

Comment:
Nice job so far👍🏼. I noticed that when the `Generate shopping list` button is clicked, an error is rendered which breaks the app, even though the shopping list is generated. Kindly fix this, as this can lead to a poor user experience. I suggest that when the button is clicked, it should lead the user to the general shopping list view so the user can see the result of his actions😉.

<img width="1434" alt="image" src="https://github.com/tsheporamantso/Recipe-App/assets/64643995/d8114061-18ae-4bb6-abf4-06d62bbd39fa">